stl/inc/ranges

```diff
@@ -2635,6 +2589,421 @@ namespace ranges {
+                && is_nothrow_move_constructible_v<sentinel_t<_Base>>) // strengthened
+                : _Last(_RANGES end(_Parent._Range)) {}
+
+            // clang-format off
+            constexpr _Sentinel(_Sentinel<!_Const> _Se)
+                noexcept(is_nothrow_constructible_v<sentinel_t<_Base>, sentinel_t<_Vw>>) // strengthened
+                requires _Const && convertible_to<sentinel_t<_Vw>, sentinel_t<_Base>>
+                : _Last(_STD move(_Se._Last)) {}
+
+            template <bool _OtherConst>
+                requires sentinel_for<sentinel_t<_Base>, _Maybe_const_iter<_OtherConst>>
+            _NODISCARD friend constexpr bool operator==(const _Iterator<_OtherConst>& _Left,
+                const _Sentinel& _Right) noexcept(noexcept(_Right._Equal(_Left))) /* strengthened */ {
+                // clang-format on
+                return _Right._Equal(_Left);
+            }
+        };
+
+    public:
+        join_view() = default;
+        constexpr explicit join_view(_Vw _Range_) noexcept(is_nothrow_move_constructible_v<_Vw>) // strengthened
+            : _Range(_STD move(_Range_)) {}
+
+        _NODISCARD constexpr _Vw base() const& noexcept(
+            is_nothrow_copy_constructible_v<_Vw>) /* strengthened */ requires copy_constructible<_Vw> {
+            return _Range;
+        }
+        _NODISCARD constexpr _Vw base() && noexcept(is_nothrow_move_constructible_v<_Vw>) /* strengthened */ {
+            return _STD move(_Range);
+        }
+
+        _NODISCARD constexpr auto begin() {
+            constexpr bool _Use_const = _Simple_view<_Vw> && is_reference_v<range_reference_t<_Vw>>;
+            return _Iterator<_Use_const>{*this, _RANGES begin(_Range)};
+        }
+
+        // clang-format off
+        _NODISCARD constexpr _Iterator<true> begin() const
+            requires input_range<const _Vw> && is_reference_v<range_reference_t<const _Vw>> {
+            // clang-format on
+            return _Iterator<true>{*this, _RANGES begin(_Range)};
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto end() {
+            if constexpr (forward_range<_Vw> && is_reference_v<range_reference_t<_Vw>>
+                    && forward_range<_InnerRng> && common_range<_Vw> && common_range<_InnerRng>) {
+                // clang-format on
+                return _Iterator<_Simple_view<_Vw>>{*this, _RANGES end(_Range)};
+            } else {
+                return _Sentinel<_Simple_view<_Vw>>{*this};
+            }
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto end() const
+            requires input_range<const _Vw> && is_reference_v<range_reference_t<const _Vw>> {
+            if constexpr (forward_range<const _Vw> && is_reference_v<range_reference_t<const _Vw>>
+                    && forward_range<range_reference_t<const _Vw>> && common_range<const _Vw>
+                    && common_range<range_reference_t<const _Vw>>) {

```

Comment:
I think this would benefit a lot from `_InnerRng<false>`

Reply:
I'd agree had you said `_InnerRng<true>` - these are `const _Vw`.